### PR TITLE
Show full Linearizer docstrings and minor improvements to the docstrings.

### PR DESCRIPTION
**Pull request by moorepants**
#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by moorepants** [comment]
@sympy/mechanics ready for review

File: sympy/physics/mechanics/linearize.py [hunk 1]
```diff
@@ -253,35 +262,42 @@ def linearize(self, op_point=None, A_and_B=False, simplify=False):
...
+
+        Notes
+        =====
+
+        Note that the process of solving with A_and_B=True is computationally
+        intensive if there are many symbolic parameters. For this reason, it
+        may be more desirable to use the default A_and_B=False, returning M, A,
+        and B. More values may then be substituted in to these matrices later
+        on. The state space form can then be found as A = P.T*M.LUsolve(A), B =
+        P.T*M.LUsolve(B), where P = Linearizer.perm_mat.
```
Comment (TJStienstra) [inline sympy/physics/mechanics/linearize.py:299]: Do we want the code like `A_and_B=True` and `A = P.T * M.LUsolve(A)` to be displayed as code?

File: sympy/physics/mechanics/linearize.py [hunk 2]
```diff
@@ -253,35 +262,42 @@ def linearize(self, op_point=None, A_and_B=False, simplify=False):
...
             Determines if returned values are simplified before return.
             For large expressions this may be time consuming. Default is False.
 
-        Potential Issues
-        ================
+        Returns
+        =======
+        M, A, B : Matrices, ``A_and_B=False``
+            Matrices from the implicit form:
+                ``[M]*[q', u']^T = [A]*[q_ind, u_ind]^T + [B]*r``
```
Comment (TJStienstra) [inline sympy/physics/mechanics/linearize.py:286]: You could in this case consider use latex. Though this is more easy to read if someone is reading it as a docstring and not in the online documentation.

**Review by tjstienstra — COMMENTED** [review]
Looks like a nice improvement making the documentation of this class clearer.

File: sympy/physics/mechanics/linearize.py [hunk 3]
```diff
@@ -11,10 +11,10 @@
 
 
 class Linearizer:
-    """This object holds the general model form for a dynamic system.
-    This model is used for computing the linearized form of the system,
-    while properly dealing with constraints leading to  dependent
-    coordinates and speeds.
+    """This object holds the general model form for a dynamic system. This
```
[the remark this answers is not in the document]
Comment (moorepants) [reply sympy/physics/mechanics/linearize.py:14]: Sure, but I think making these docstrings compliant with the new style guide or other extra fixes can be in a new PR.

File: sympy/physics/mechanics/linearize.py (hunk 1 above)
Comment (moorepants) [reply sympy/physics/mechanics/linearize.py:299]: I didn't edit it. Can also be done in a new PR.

**Comment by TJStienstra** [comment]
Okay then it looks good to me.

**Comment by moorepants** [comment]
We probably need to have a go over the entire mechanics/vector docs to get them improved and aligned with the style guide. I agree with all your comments and will try to address them in further PRs.

**Pull request merged**